packages/wonder-blocks-core/components/unique-id-provider.test.js

```diff
@@ -124,4 +125,28 @@ describe("UniqueIDProvider", () => {
             expect(children.mock.calls[2][0]).toBe(children.mock.calls[1][0]);
         });
     });
+
+    describe("inside a NoSSR", () => {
+        test("it should pass an id to its children", () => {
+            // Arrange
+            const foo = jest.fn(() => null);
+            const nodes = (
+                <NoSSR>
+                    {() => (
+                        <UniqueIDProvider>
+                            {(ids) => foo(ids.get(""))}
+                        </UniqueIDProvider>
+                    )}
+                </NoSSR>
+            );
+            const wrapper = mount(nodes);
+
+            // Act
+            wrapper.instance().forceUpdate();
```

Comment:
Should we verify that `foo` is not called before the `forceUpdate` just to ensure that we're testing what we think we are?

Reply:
I can add that check.

Reply:
No, I think you can drop the `forceUpdate` and change the `// Act` to just be `mount(nodes);`.

Since the provider is wrapped in the `NoSSR` component, my guess is that the `setState` in the `NoSSR` component `componentDidMount` is making the second render occur, and causing the provider to do its render.

Reply:
I'm going to remove this from the test since it isn't necessary.